packages/contracts/test/sdk/universe/single-token/erc721.test.ts

```diff
@@ -0,0 +1,763 @@
+import { Contract } from "@ethersproject/contracts";
+import { parseEther } from "@ethersproject/units";
+import * as Common from "@reservoir0x/sdk/src/common";
+import * as Universe from "@reservoir0x/sdk/src/Universe";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { getChainId, reset, setupNFTs } from "../../../utils";
+import { AssetClass, OrderSide } from "@reservoir0x/sdk/src/universe/types";
+import { BigNumber, constants, utils } from "ethers";
+
+describe("Universe - SingleToken Erc721", () => {
+  const chainId = getChainId();
+
+  let deployer: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let bob: SignerWithAddress;
+  let charlie: SignerWithAddress;
+  let dan: SignerWithAddress;
+
+  let erc721: Contract;
+
+  beforeEach(async () => {
+    [deployer, alice, bob, charlie, dan] = await ethers.getSigners();
+
+    ({ erc721 } = await setupNFTs(deployer));
+  });
+
+  afterEach(reset);
+
+  it("Build and fill ERC721 WETH buy order no revenue splits", async () => {
+    const seller = alice;
+    const buyer = bob;
+    const price = parseEther("1");
+    const soldTokenId = 0;
+
+    const weth = new Common.Helpers.Weth(ethers.provider, chainId);
+
+    // Mint weth to buyer
+    await weth.deposit(buyer, price);
+
+    // Approve the exchange contract for the buyer
+    await weth.approve(buyer, Universe.Addresses.Exchange[chainId]);
+
+    const buyerBalanceBefore = await weth.getBalance(seller.address);
+    const sellerBalanceBefore = await weth.getBalance(buyer.address);
+
+    // Mint erc721 to seller
+    await erc721.connect(seller).mint(soldTokenId);
+
+    const nft = new Common.Helpers.Erc721(ethers.provider, erc721.address);
+
+    // Approve the transfer manager
+    await nft.approve(seller, Universe.Addresses.Exchange[chainId]);
+
+    const exchange = new Universe.Exchange(chainId);
+    const daoFee = await exchange.getDaoFee(seller.provider!);
+
+    const builder = new Universe.Builders.SingleToken(chainId);
+    // Build sell order
+    const buyOrder = builder.build({
+      hash: "",
+      type: "UNIVERSE_V1",
```

Comment:
Do we need the type in the order details? It's not needed for filling.

Reply:
It's not needed for onchain filling but might be needed in case the offchain order structure changes and we need to differentiate order versions

Reply:
Ok then it makes sense to keep it